django/template/loader_tags.py

```diff
@@ -257,6 +257,14 @@ def construct_relative_path(
         # relative path.
         return relative_name
 
+    if not current_template_name:
+        if new_name.startswith("../"):
+            raise TemplateSyntaxError(
+                "The relative path %s points outside the "
+                "template directory." % (relative_name)
+            )
+        return relative_name
```

Comment:
I don't think it makes sense for these 2 cases to be treated differently
I would consider doing:
```python
    if not new_name.startswith(("./", "../")) or not current_template_name:
        return relative_name
```
Then just let these cases be picked up by the template does not exist error